pkg/types/vsphere/conversion/installconfig.go

```diff
@@ -71,38 +150,89 @@ func ConvertInstallConfig(config *types.InstallConfig) error {
 
 		platform.FailureDomains[0].Topology.Datacenter = platform.DeprecatedDatacenter
 		platform.FailureDomains[0].Topology.ResourcePool = platform.DeprecatedResourcePool
-		platform.FailureDomains[0].Topology.ComputeCluster = setComputeClusterPath(platform.DeprecatedCluster, platform.DeprecatedDatacenter)
+		platform.FailureDomains[0].Topology.ComputeCluster, err = setComputeClusterPath(finder, platform.DeprecatedCluster, platform.DeprecatedDatacenter)
+
+		if err != nil {
+			return err
+		}
 		platform.FailureDomains[0].Topology.Networks = make([]string, 1)
 		platform.FailureDomains[0].Topology.Networks[0] = platform.DeprecatedNetwork
-		platform.FailureDomains[0].Topology.Datastore = setDatastorePath(platform.DeprecatedDefaultDatastore, platform.DeprecatedDatacenter)
-		platform.FailureDomains[0].Topology.Folder = setFolderPath(platform.DeprecatedFolder, platform.DeprecatedDatacenter)
+		platform.FailureDomains[0].Topology.Datastore, err = setDatastorePath(finder, platform.DeprecatedDefaultDatastore, platform.DeprecatedDatacenter)
+
+		if err != nil {
+			return err
+		}
+		platform.FailureDomains[0].Topology.Folder, err = setFolderPath(finder, platform.DeprecatedFolder, platform.DeprecatedDatacenter)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// ConvertInstallConfig modifies a given platform spec for the new requirements.
+func ConvertInstallConfig(config *types.InstallConfig) error {
+	platform := config.Platform.VSphere
+
+	fixNoVCentersScenario(platform)
+	finder, err := getFinder(platform.VCenters[0].Server, platform.VCenters[0].Username, platform.VCenters[0].Password)
+	if err != nil {
+		return err
+	}
+	err = fixTechPreviewZonalFailureDomainsScenario(platform, finder)
+	if err != nil {
+		return err
+	}
+	err = fixLegacyPlatformScenario(platform, finder)
+	if err != nil {
+		return err
 	}
 
 	return nil
 }
 
-func setComputeClusterPath(cluster, datacenter string) string {
+func setComputeClusterPath(finder *find.Finder, cluster, datacenter string) (string, error) {
 	if cluster != "" && !strings.HasPrefix(cluster, "/") {
 		localLogger.Warnf("computeCluster as a non-path is now deprecated; please use the form: /%s/host/%s", datacenter, cluster)
-		return fmt.Sprintf("/%s/host/%s", datacenter, cluster)
+
+		clusterPathFind := path.Join("/", datacenter, "host", "...")
+		clusterPath := path.Join("/", datacenter, "host", cluster)
+
+		// There is no connection to vCenter, use unconnected method for generating path
+		if finder == nil {
```

Comment:
How about a log message here, so we can tell which path was taken when debugging?

Reply:
Sure, I will make that change.

Reply:
Since this message is the same as the one on L209, we still wouldn't be able to tell from the logs which code path was taken (assuming log level is higher than debug and L207 is silent).

I dig the idea of that message being emitted in any scenario where we enter the condition on L192. I'm thinking this one should indicate that we're "generating a guess at the correct path because no vcenter connection is available".